token-lending/program/src/processor.rs

```diff
@@ -1714,6 +1724,69 @@ fn unpack_mint(data: &[u8]) -> Result<Mint, LendingError> {
     Mint::unpack(data).map_err(|_| LendingError::InvalidTokenMint)
 }
 
+fn get_pyth_product_attribute<'a, T>(ite: &mut T) -> String
+where
+    T: Iterator<Item = &'a u8>,
+{
+    let mut len = *ite.next().unwrap() as usize;
+    let mut val = String::with_capacity(len);
+    while len > 0 {
+        val.push(*ite.next().unwrap() as char);
+        len -= 1;
+    }
+    val
+}
+
+fn get_pyth_price(
+    pyth_price_account_info: &AccountInfo,
+    clock: &Clock,
+) -> Result<u64, ProgramError> {
+    let pyth_price_data = &pyth_price_account_info.try_borrow_data()?;
+    let pyth_price = pyth_client::cast::<pyth_client::Price>(pyth_price_data);
+
+    match pyth_price.ptype {
+        pyth_client::PriceType::Price => {}
+        _ => {
+            msg!("Oracle price type is invalid");
+            return Err(LendingError::InvalidOracleConfig.into());
+        }
+    }
+
+    if pyth_price.valid_slot < clock.slot - SLOTS_PER_MINUTE {
```

Comment:
a minute is a long time, I think we need this to be shorter. Do we have any idea yet how often pyth feeds will be updated?

Reply:
> The pyth program accumulates prices with respect to whatever the current slot is inside the solana node. This is called the "valid_slot" above. As soon as the slot ticks forward by one, the pyth program computes the aggregate price and publishes it with respect to the new "publish_slot" and starts repeating the process with a new "valid_slot".

Anecdotally, they seem to be updated very rapidly. I will make this quite short and see how it behaves.

Reply:
https://explorer.solana.com/address/5mkqGkkWSaSk2NL9p4XptwEQu4d5jFTJiurbbzdqYexF?cluster=devnet (Pyth program on devnet) shows the updates happening very frequently. I've shortened this to 5 slots for now.